java/arcs/core/data/Annotation.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+/** A class containing an annotation. */
+data class Annotation(val name: String, val params: Map<String, AnnotationParam>) {
+    companion object {
+        private val ANNOTATION_PATTERN = "^([a-zA-Z]+)(\\((.+)\\))?$".toRegex()
+        private val PARAM_PATTERN = "^([a-zA-Z]+):(.*)$".toRegex()
+
+        fun fromString(str: String): List<Annotation> {
+            val tokens = str.split("@")
+            return tokens.drop(1).map {
```

Comment:
I think there's a lot of bugs hidden here:
* what happens if there's no `@` in `str`?
* what happens if there's an `@` inside a string param in the annotation?

Hence my question above, when will this function be used? This seems like a job for a proper parser library.

If you need this now, could you get away with only parsing a single annotation at a time? That would be easier, I think.

Reply:
refactored this method to parse a single annotation, starting with `@`